firmware/dev/SSM/src/io/io_dac.c

```diff
@@ -0,0 +1,63 @@
+#include "io_dac.h"
+
+#include "app_utils.h"
+#include "hw_spis.h"
+#include "hw_gpios.h"
+
+#include <assert.h>
+
+#define V_REF (4.576f)
+#define RESOLUTION (12)
+
+bool io_dac_writeReg(const DACChannel reg_addr, const float volts)
+{
+    const uint16_t data = (uint16_t)(volts * (1 << RESOLUTION) / V_REF) & 0xFFF;
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wconversion"
+    struct __attribute__((packed))
+    {
+        // byte 1
+        uint8_t address : 4;
+        uint8_t cmd : 4;
+
+        // byte 2
+        uint8_t data_1;
+
+        // byte 3
+        uint8_t data_2;
+    } tx_msg = { .cmd = 0x3, .address = reg_addr, .data_1 = 0xff & data >> 4, .data_2 = 0xf & data };
```

Comment:
note that this commands both turns on the DAC and writes to it. there might be value in separating it? but I don't see it